share/trick/trickops/TrickWorkflow.py

```diff
@@ -1380,20 +1474,25 @@ def connect():
                     return
                 except socket.timeout:
                     pass
+                # If a SingleRun job terminates before the thread can connect to the
+                # variable server, Trick's variable_server module throws an IOError
+                # with the message: The remote endpoint has closed the connection
+                except IOError as e:
```

Comment:
FYI @jdeans289 this resolves the `IOError("The remote endpoint has closed the connection")` problem we see when the variable server tries to connect to a sim that has already terminated.  In case you're curious, the easiest way to replicate this is by instrumenting one of the test sims (I used `SIM_ball_L1`) with this type of "long-initialization, short runtime" setup:

```python
trick.exec_set_terminate_time(1.0)
trick.add_read(0.0, "import time; time.sleep(10)")
```
This gives TrickOps enough time to attempt a connection but not enough time for the handshake to complete.